Code/GraphMol/RGroupDecomposition/RGroupDecompData.h

```diff
@@ -31,41 +32,35 @@ struct RGroupDecompData {
   std::map<int, std::vector<int>> userLabels;
 
   std::vector<int> processedRlabels;
-  std::map<int, boost::shared_ptr<RWMol>> labelledCores;
 
   std::map<int, int> finalRlabelMapping;
 
   RGroupDecompData(const RWMol &inputCore,
                    RGroupDecompositionParameters inputParams)
-      : cores(),
-        newCores(),
-        newCoreLabel(EMPTY_CORE_LABEL),
-        params(std::move(inputParams)) {
-    cores[0] = inputCore;
+      : params(std::move(inputParams)) {
+    cores[0] = RCore(inputCore);
     prepareCores();
   }
 
   RGroupDecompData(const std::vector<ROMOL_SPTR> &inputCores,
                    RGroupDecompositionParameters inputParams)
-      : cores(),
-        newCores(),
-        newCoreLabel(EMPTY_CORE_LABEL),
-        params(std::move(inputParams)) {
+      : params(std::move(inputParams)) {
     for (size_t i = 0; i < inputCores.size(); ++i) {
-      cores[i] = *inputCores[i].get();
+      cores[i] = RCore(*inputCores[i]);
     }
 
     prepareCores();
   }
 
   void prepareCores() {
-    size_t idx = 0;
-    for (auto coreIt = cores.begin(); coreIt != cores.end(); ++coreIt, ++idx) {
-      RWMol *alignCore = coreIt->first ? &cores[0] : nullptr;
-      CHECK_INVARIANT(params.prepareCore(coreIt->second, alignCore),
+    for (auto &core : cores) {
+      RWMol *alignCore = core.first ? cores[0].core.get() : nullptr;
+      CHECK_INVARIANT(params.prepareCore(*core.second.core, alignCore),
```

Comment:
There is maybe a case for moving this logic into RCore

```
RCore::prepareCore (const &RGroupDecompositionParameters params, const RWMol& alignCore)
```

or perphaps prepareCore should be:

```
RGroupDecompositionParameters::prepareCore(RCore &, const RWMol &alignCore)
```

Reply:
@bp-kelley Hi Brian, thanks for reviewing this. As soon as this is merged I will move on to the remainder of the PR in the other branch (@greglandrum maybe you wish to have a quick look before merging?).